### ISSUE TITLE
Damage Plugin and EffectDamage

### ISSUE THREAD

**Issue by rinehardt** [nwnxee/unified#294]
Anyone else having strange behavior with EffectDamage when modifying damage data struct values? It seems to be adding the new amount to the existing amount and then using that number for both damage values. For example, setting 5 acid damage on a struct with 30 fire damage causes it to deal 35 fire and 35 acid damage.

I've attached a zip containing a screenshot of my damage logs and a test module with a few scripts to help demonstrate.

[damage_plugin_bug_report.zip](https://github.com/nwnxee/unified/files/2776132/damage_plugin_bug_report.zip)

**Comment by mtijanic** [comment]
Yep, this will not work unfortunately. If it goes through EffectDamage(), it has to be a single damage type.
As far as I can see, when you add 5 acid to 30 fire, it will only do 35 fire, but the message will report 35fire+35acid. However, I think only 35 hp is reduced.

The only easy way to solve this that I see is to flag the damage as having been done in combat (e.g. from an attack), but that will mean it'll bypass damage reduction/immunity, as for combat damage that has been already handled.

Incidentally, the damage reported here is post-DR for combat damage, but pre-DR for EffectDamage() stuff.

I don't think this is gonna get fixed. But we do need a huge instruction manual how to use this at all.

**Comment by henningkoehlernz** [comment]
You can deal multiple types of damage with the new [DealDamage](https://github.com/nwnxee/unified/pull/420) function.

**Comment by Daztek** [comment]
Appears to have been implemented/fixed!
